app/models/host.rb

```diff
@@ -18,6 +18,10 @@ class Host < ApplicationRecord
   has_many :policies, through: :policy_hosts
   has_many :assigned_profiles, through: :policies, source: :profiles
 
+  scope :os_versions, lambda { |hosts|
+    select(:system_profile).where(id: hosts).map(&:os_version).uniq
+  }
```

Comment:
This isn't how scopes are typically used. They should return another ActiveRecord::Relation. You may want a class method instead. In either case, you can pluck the OS versions directly using SQL:

```rb
pluck(%w(system_profile->'operating_system'->'major' system_profile->'operating_system'->'minor'))
```